lib/lintSource.js

```diff
@@ -119,7 +119,7 @@ function lintPostcssResult(
     postcssResult.stylelint.customMessages[ruleName] = _.get(secondaryOptions, "message")
 
     const performRule = Promise.resolve().then(() => {
-      ruleFunction(primaryOption, secondaryOptions)(postcssRoot, postcssResult)
+      ruleFunction(primaryOption, secondaryOptions)(postcssRoot, postcssResult, config)
```

Comment:
Adding a third argument actually opens up a wide range of possibilities ... Instead of limiting that range by making this string the config object, maybe the argument should an object with a `config` property. If we find other uses for this argument (e.g. we could attach the internal stylelint API to it), we could then add them to the object without a breaking change or more arguments.